torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -2688,17 +2740,11 @@ bool insertProfileIValue(ProfilingRecord* pr, Node* node, size_t offset) {
     return true;
   }
 
-  /*
-  // TODO: Enable view in parser by detecting non-alias view operation
   static auto view_schema =
       getOperatorForLiteral(
-          "aten::view(Tensor(a) self, int[] size) -> Tensor(a)")
+          "prim::view_copy(Tensor(a) self, int[] size) -> Tensor(a)")
           ->schema();
-  static auto reshape_schema =
-      getOperatorForLiteral(
-          "aten::reshape(Tensor(a) self, int[] shape) -> Tensor(a)")
-          ->schema();
-  if (node->matches(view_schema) || node->matches(reshape_schema)) {
+  if (node->matches(view_schema)) {
```

Comment:
Using `profileSize` here doesn't seem right.

It's not your fault, `profileSize` should really be renamed as `profile_reduction_size`. The guard logic is to simply check that the size at runtime is of the same rank and elements in sizes that is equal to 1 is not changed.

So for `view` op, the guarding logic is going to be very different. Since it combines a broadcast as well as a reduction. To properly guard that, we'll also need to look at the input tensor as well.